### PR TITLE
Calculated elements are not searchable by default

### PR DESCRIPTION
**Pull request by vmikhailenko**
In Java runtime, the calculated elements with "on-read" (more precisely, ones that annotated with `@Core.Computed` ) semantics are excluded from the search by default. 
This is the case for Node.js as well, but the condition is a little different. 

Java excludes the virtual elements from the search even if they are explicitly annotated. Should we document that as well?

### CODE REVIEW

File: guides/providing-services.md [hunk 1]
```diff
@@ -449,6 +449,8 @@ entity Books { ... }
 
 Searches all elements of type `String` excluding the element `isbn`, which leaves the `title` and `descr` elements to be searched.
 
+You can extend the search to calculated elements which are not searchable by default.
```
Comment (agoerler) [inline guides/providing-services.md:452]:
```suggestion
You can extend the search to calculated elements, which are not searchable by default.
```

File: guides/providing-services.md [hunk 2]
```diff
@@ -416,7 +416,7 @@ That would basically search for occurrences of `"Heights"` in all text fields of
 
 #### Using the `@cds.search` Annotation {#using-cds-search-annotation}
 
-By default all elements of type `String` of an entity are searched. Yet, sometimes you may want to deviate from this default and specify a different set of searchable elements, or to extend the search to associated entities. Use the `@cds.search` annotation to do so. The general usage is:
+By default all concrete elements of type `String` of an entity are searched. Yet, sometimes you may want to deviate from this default and specify a different set of searchable elements, or to extend the search to associated entities. Use the `@cds.search` annotation to do so. The general usage is:
```
Comment (agoerler) [inline guides/providing-services.md:419]:
```suggestion
By default only elements of type `String` of an entity are searched, provided they are not [calculated](../cds/cdl#calculated-elements) or [virtual](../cds/cdl#virtual-elements). Yet, sometimes you may want to deviate from this default and specify a different set of searchable elements, or to extend the search to associated entities. Use the `@cds.search` annotation to do so. The general usage is:
```

**Review by agoerler — APPROVED** [review]

**Comment by agoerler** [comment]
@johannes-vogel - please review and check wether is also applies to CAP Node.js

**Review by danjoa — APPROVED** [review]
👍 looks good

**Review by agoerler — APPROVED** [review]

**Review by smahati — APPROVED** [review]

**Pull request merged**